Automattic/newspack-ios
# Issues/91 share extension part 1

Refs #91 

This PR lays part of the groundwork to support sharing photos via a Share Extension.  There are two main areas of interest.
- First, an App Group is added along with relevant entitlements.  This allows the Share Extension and the main app to share info via UserDefaults and a group directory.
- Second, some code is reorganized and refactored into an internal framework for sharing between the main app and the share extension.

I'd planned to do a bit more but the changes to implement the framework are rather large (if mostly mundane/boilerplate) so now is a good time to pause for a PR. 

To test: 
Confirm all tests pass.
Confirm you can build and run the app.
Confirm you can build the share extension.
Confirm there are no warnings related to the framework changes.

@jleandroperez can I trouble you for a peek at your leisure?
